Panels/Viewport.js

```diff
@@ -0,0 +1,235 @@
+import classnames from 'classnames';
+import {forward, handle} from '@enact/core/handle';
+import Spotlight from '@enact/spotlight';
+import Pause from '@enact/spotlight/Pause';
+import ViewManager, {shape, SlideBottomArranger as VerticalArranger, SlideRightArranger as HorizontalArranger} from '@enact/ui/ViewManager';
+import invariant from 'invariant';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import SharedStateDecorator, {SharedState} from './SharedStateDecorator';
+
+import css from './Panels.module.less';
+
+/**
+ * The container for a set of Panels
+ *
+ * @class Viewport
+ * @memberof agate/Panels
+ * @private
+ */
+const ViewportBase = class extends React.Component {
+	static displayName = 'Viewport'
+
+	static contextType = SharedState
+
+	static propTypes = /** @lends agate/Panels.Viewport.prototype */ {
+
+		/**
+		 * A function that generates a globally-unique identifier for a panel index
+		 *
+		 * @type {Function}
+		 * @required
+		 */
+		generateId: PropTypes.func.isRequired,
+
+		/**
+		 * Set of functions that control how the panels are transitioned into and out of the
+		 * viewport
+		 *
+		 * @type {Arranger}
+		 */
+		arranger: shape,
+
+		/**
+		 * Panels to be rendered
+		 *
+		 * @type {Panel}
+		 */
+		children: PropTypes.node,
+
+		/**
+		 * Duration of the animation (in ms) when transitioning between `Panel` components.
+		 *
+		 * @type {Number}
+		 * @default 500
+		 * @public
+		 */
+		duration: PropTypes.number,
+
+		/**
+		 * Index of the active panel
+		 *
+		 * @type {Number}
+		 * @default 0
+		 */
+		index: PropTypes.number,
+
+		/**
+		 * Disable panel transitions
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 */
+		noAnimation: PropTypes.bool,
+
+		/**
+		 * Direction of the animation when transitioning between `Panel` components.
+		 *
+		 * This is only applied when no arranger has been set. When using `'horizontal'`, a `SlideRightArranger` is used.
+		 * When using `'vertical'`, a `SlideBottomArranger` is used.
+		 *
+		 * Valid values are:
+		 * * `'horizontal'`, and
+		 * * `'vertical'`.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		orientation: PropTypes.string
```

Comment:
WDYT of removing this prop and only using the `arranger` prop. It doesn't add a lot of value as only a proxy for selecting the arrangers?

Reply:
History: It was added specifically for that reason, to be a simpler way to adjust Panels with just a single string prop change, rather than importing multiple arrangers at app level and swapping them out during runtime. With the CES demo, the app needed to be able to conveniently switch between orientations, so it made sense at the time. It's probably not the best approach.